src/internal/communication.ts

```diff
@@ -0,0 +1,472 @@
+    Communication.parentMessageQueue = [];
+    Communication.childWindow = null;
+    Communication.childOrigin = null;
+    Communication.childMessageQueue = [];
+    Communication.nextMessageId = 0;
+    Communication.callbacks = {};
+  }
+
+  /**
+   * Send a message to parent. Uses nativeInterface on mobile to communicate with parent context
+   */
+  public static sendMessageToParent(actionName: string): void;
+  /**
+   * Send a message to parent. Uses nativeInterface on mobile to communicate with parent context
+   */
+  public static sendMessageToParent(actionName: string, args: any[]): void;
+  /**
+   * Send a message to parent. Uses nativeInterface on mobile to communicate with parent context
+   */
+  public static sendMessageToParent(actionName: string, callback: Function): void;
+  /**
+   * Send a message to parent. Uses nativeInterface on mobile to communicate with parent context
+   */
+  public static sendMessageToParent(actionName: string, args: any[], callback: Function): void;
+  public static sendMessageToParent(actionName: string, argsOrCallback?: any[] | Function, callback?: Function): void {
+    let args: any[] | undefined;
+    if (argsOrCallback instanceof Function) {
+      callback = argsOrCallback;
+    } else if (argsOrCallback instanceof Array) {
+      args = argsOrCallback;
+    }
+
+    const targetWindow = Communication.parentWindow;
+    const request = Communication.createMessageRequest(actionName, args);
+    if (GlobalVars.isFramelessWindow) {
+      if (Communication.currentWindow && Communication.currentWindow.nativeInterface) {
+        (Communication.currentWindow as ExtendedWindow).nativeInterface.framelessPostMessage(JSON.stringify(request));
+      }
+    } else {
+      const targetOrigin = Communication.getTargetOrigin(targetWindow);
+
+      // If the target window isn't closed and we already know its origin, send the message right away; otherwise,
+      // queue the message and send it after the origin is established
+      if (targetWindow && targetOrigin) {
+        targetWindow.postMessage(request, targetOrigin);
+      } else {
+        Communication.getTargetMessageQueue(targetWindow).push(request);
+      }
+    }
+
+    if (callback) {
+      Communication.callbacks[request.id] = callback;
+    }
+  }
+
+  public static registerHandler(name: string, handler: Function): void {
+    Communication.handlers[name] = handler;
+  }
+
+  public static removeHandler(name: string): void {
+    delete Communication.handlers[name];
+  }
+
+  private static handleStartConversation(
+    subEntityId: string,
+    conversationId: string,
+    channelId: string,
+    entityId: string,
+  ): void {
+    if (GlobalVars.onStartConversationHandler) {
+      GlobalVars.onStartConversationHandler({
+        subEntityId: subEntityId,
+        conversationId: conversationId,
+        channelId: channelId,
+        entityId: entityId,
+      });
+    }
+  }
+
+  private static handleCloseConversation(
```

Comment:
There are still using GlobalVars.onStartConversationHandler. May be we can move these back to InternalAPIs file.

Reply:
See PR #503. It takes care of handlers